curated_transformers/models/hf_hub.py

```diff
@@ -86,3 +80,71 @@ def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True):
         order to be an abstract base class.
         """
         ...
+
+
+def _get_model_config_filepath(name: str, revision: str) -> str:
+    try:
+        return hf_hub_download(
+            repo_id=name, filename=HF_MODEL_CONFIG, revision=revision
+        )
+    except:
+        raise ValueError(
+            f"Couldn't find a valid config for model `{name}` "
+            f"(revision `{revision}`) on HuggingFace Model Hub"
+        )
+
+
+def _get_model_checkpoint_filepaths(name: str, revision: str) -> Iterable[str]:
+    # Attempt to download a non-sharded checkpoint first.
```

Comment:
I was wondering if we don't want to try getting a sharded model first. If haven't looked into this yet, but I think that one of the benefits of shard loading is that in the future we could e.g. quantize the parameters by shard when loading large models. As a result we never need to have the full FP32 model in memory.

Reply:
Yeah, that was my thinking too. That being said, the order in which we check/get the filepaths shouldn't matter as sharded and non-sharded checkpoints are mutually exclusive (as far as I can tell) - The `python_model.bin` file is only present in the case of the latter.